main/src/ui/encoder.cpp

```diff
@@ -26,17 +26,56 @@ void onClickZeroMachineXY(EncoderButton &eb) {
 	drawCenteredText("Zeroing Machine XY...", 2);
 	machineZeroXY();
 	state = MACHINE_XY_ZERO;
-	drawCenteredText("Zero Workspace Z", 2);
-	encoder.setClickHandler(onClickZeroWorkspaceZ);
+    StartZeroWorkspace();
+}
+
+void StartZeroWorkspace()
+{
+    drawCenteredText("Zero Workspace Z\nHold L+R for\nauto touch", 2);
+    encoder.setClickHandler(onClickZeroWorkspaceZ);
+    handleButtons.setButtonsHoldHandler(onHoldAutoTouch);
+    handleButtons.enable(true);
+}
+
+void onHoldAutoTouch(HandleButtons &btn) {
+	drawCenteredText("Auto touch...", 2);
+	autoTouchWorkspaceZ();
+	handleButtons.setLeftClickHandler(onClickCancelAutoTouch);
+	handleButtons.setRightClickHandler(onClickAcceptAutoTouchH);
+	encoder.setClickHandler(onClickAcceptAutoTouchE);
+	drawCenteredText("Touched!", 2);
 }
 
 void onClickZeroWorkspaceZ(EncoderButton &eb) {
+	handleButtons.enable(false);
+	handleButtons.clearHandlers();
 	drawCenteredText("Zeroing Workspace Z...", 2);
 	workspaceZeroZ();
 	state = WORKSPACE_Z_ZERO;
 	encoderSetThickness();
 }
 
+void onClickCancelAutoTouch(HandleButtons &btn) {
+	handleButtons.enable(false);
+	handleButtons.clearHandlers();
+	StartZeroWorkspace();
+}
+
+void onClickAcceptAutoTouch() {
+	handleButtons.enable(false);
+	acceptAutoTouchWorkspaceZ();
+	state = WORKSPACE_Z_ZERO;
+	encoderSetThickness();
+}
+
+void onClickAcceptAutoTouchE(EncoderButton &eb) {
+	onClickAcceptAutoTouch();
+}
+
+void onClickAcceptAutoTouchH(HandleButtons &eb) {
```

Comment:
I assume this should be a button caller `&btn`?

```suggestion
void onClickAcceptAutoTouchH(HandleButtons &btn) {
```